src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -2649,6 +2649,12 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 break;
             }
 
+            case NI_Sve2_BitwiseClearXor:
+                assert(targetReg == op1Reg);
```

Comment:
you need to add `movprfx` if `targetReg != op1Reg` something like:

```c++
                    if (targetReg != op1Reg)
                    {
                        assert(targetReg != op2Reg);

                        GetEmitter()->emitInsSve_R_R(INS_sve_movprfx, EA_SCALABLE, targetReg, op1Reg);
                    }
````

See how we handle `TrigonometricMultiplyAddCoefficient` in this file.

Reply:
And it seems that's why we are seeing test failure too:

```
Assert failure(PID 60 [0x0000003c], Thread: 60 [0x003c]): Assertion failed 'targetReg == op1Reg' in 'JIT.HardwareIntrinsics.Arm._Sve2.SimpleTernaryOpTest__Sve2_BitwiseClearXor_sbyte:RunLclVarScenario_UnsafeRead():this' during 'Generate code' (IL size 113; hash 0xa6f94412; FullOpts)

    File: /__w/1/s/src/coreclr/jit/hwintrinsiccodegenarm64.cpp:2653
    Image: /root/helix/work/correlation/corerun
```

Reply:
The need for `movprfx` for RMW handling is notably why x64 has helpers like `emitIns_SIMD_R_R_R`: https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/emitxarch.cpp#L9766-L9791

Such helpers allow codegen to mostly ignore the concept and just thing in terms of number of input/outputs instead. It allows for emit to centralize the handling instead. This makes it much simpler to update if we have new considerations in the future, to avoid accidentally missing the handling in various locations, etc.

Reply:
I agree. The `movprfx` logic needs to be centralized for sure.

Reply:
https://github.com/dotnet/runtime/issues/115508